test/TestUtils/TestProjectFixture.cs

```diff
@@ -0,0 +1,290 @@
+
+            var buildArgs = new List<string>();
+            if (runtime != null)
+            {
+                buildArgs.Add("--runtime");
+                buildArgs.Add(runtime);
+            }
+
+            if (framework != null)
+            {
+                buildArgs.Add("--framework");
+                buildArgs.Add(framework);
+            }
+
+            if (outputDirectory != null)
+            {
+                buildArgs.Add("-o");
+                buildArgs.Add(outputDirectory);
+            }
+
+            dotnet.Build(buildArgs.ToArray())
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .EnsureSuccessful();
+
+            _testProject.LoadOutputFiles();
+
+            return this;
+        }
+
+        public TestProjectFixture PublishProject(
+            DotNetCli dotnet = null,
+            string runtime = null,
+            string framework = "netcoreapp1.0",
+            string outputDirectory = null)
+        {
+            dotnet = dotnet ?? _sdkDotnet;
+            outputDirectory = outputDirectory ?? _testProject.OutputDirectory;
+            _testProject.OutputDirectory = outputDirectory;
+
+            var publishArgs = new List<string>();
+            if (runtime != null)
+            {
+                publishArgs.Add("--runtime");
+                publishArgs.Add(runtime);
+            }
+
+            if (framework != null)
+            {
+                publishArgs.Add("--framework");
+                publishArgs.Add(framework);
+            }
+
+            if (outputDirectory != null)
+            {
+                publishArgs.Add("-o");
+                publishArgs.Add(outputDirectory);
+            }
+
+            dotnet.Publish(publishArgs.ToArray())
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .EnsureSuccessful();
+
+            _testProject.LoadOutputFiles();
+
+            return this;
+        }
+
+        public TestProjectFixture RestoreProject(DotNetCli dotnet = null)
+        {
+            dotnet = dotnet ?? _sdkDotnet;
+
+            dotnet.Restore()
+                .WorkingDirectory(_testProject.ProjectDirectory)
+                .Execute()
+                .EnsureSuccessful();
+
+            return this;
+        }
+
+        public TestProjectFixture EnsureRestored(DotNetCli dotnet = null)
+        {
+            dotnet = dotnet ?? _sdkDotnet;
+
+            if ( ! _testProject.IsRestored())
+            {
+                RestoreProject(dotnet: dotnet);
+            }
+
+            return this;
+        }
+
+        public TestProjectFixture Copy()
+        {
+            return new TestProjectFixture(this);
+        }
+
+        public TestProjectFixture MoveDepsJsonToSubdirectory()
```

Comment:
I see what is happening here, it didn't make sense before but it does now.

I think these "MoveXXXToSubdirectory" methods should be contained in the test themselves, and not the "test framework".  These methods are only needed for specific test cases, and are not applicable across most of the tests.  Moving these methods out will keep this class more "cohesive". It means this class is only responsible for "framework" type of methods.  These methods don't really belong with the "Restore/Build/Publish" methods.

---
In reply to: [66107124](https://github.com/dotnet/core-setup/pull/97#discussion_r66107124) [](ancestors = 66107124)